src/cow-react/common/hooks/useParameterizedMainMenu.ts

```diff
@@ -0,0 +1,20 @@
+import { BasicMenuLink, isBasicMenuLink, MAIN_MENU, MenuTreeItem } from 'constants/mainMenu'
+import { useParameterizeLimitOrdersRoute } from 'cow-react/modules/limitOrders/hooks/useParameterizeLimitOrdersRoute'
+import { Routes } from 'constants/routes'
+import { useMemo } from 'react'
+
+export function useParameterizedMainMenu(): MenuTreeItem[] {
+  const limitOrdersRouteWithParams = useParameterizeLimitOrdersRoute()
+
+  return useMemo(() => {
+    const copy: MenuTreeItem[] = JSON.parse(JSON.stringify(MAIN_MENU))
+
+    const limitOrdersRoute = copy.filter<BasicMenuLink>(isBasicMenuLink).find((item) => item.url === Routes.LIMIT_ORDER)
```

Comment:
If this is a `common` component, why this dependency with with LIMIT ORDERS? 

Logic for getting params from the URL in limit orders, should be in the limit order page, or somewhere in the swap module, right?

Reply:
https://github.com/cowprotocol/cowswap/pull/1110#issuecomment-1263237581

Unfortunately, here I have to violate the principle of dependency inversion :(
This can be avoided by creating an intermediate step, something like a store.
What do you think?

Reply:
Fixed here: https://github.com/cowprotocol/cowswap/pull/1134